module/zfs/dbuf_stats.c

```diff
@@ -134,7 +134,8 @@ dbuf_stats_hash_table_data(char *buf, size_t size, void *data)
 
 	ASSERT3S(dsh->idx, >=, 0);
 	ASSERT3S(dsh->idx, <=, h->hash_table_mask);
-	memset(buf, 0, size);
+	if (size)
+		buf[0] = 0;
```

Comment:
This seems odd; if the size is non-zero, we terminate it immediately rather than at `size - 1`?

Reply:
Yes. `size` is the size of the buffer. We want to terminate it immediately only as a precaution against there being zero buffers. If there are any buffers than the snprintf will NULL terminate it. NULL terminating it as `size-1` would leave you with arbitrary garbage in the output.

Reply:
Ah, this makes sense- thanks!